controller.py

```diff
@@ -144,8 +144,54 @@ def update_url(time=last_update_time, page=None):
 
         return feed_response(update_feed)
 
+    def add_items(self, collection_details):
+        """Adds identifiers to a Collection's catalog"""
+        client = authenticated_client_from_request(self._db)
+        if isinstance(client, ProblemDetail):
+            return client
+
+        collection, ignore = Collection.from_metadata_identifier(
+            self._db, collection_details
+        )
+
+        urns = request.args.getlist('urn')
+        messages = []
+        for urn in urns:
+            message = None
+            identifier = None
+            try:
+                identifier, ignore = Identifier.parse_urn(
+                    self._db, urn
+                )
+            except Exception as e:
+                identifier = None
+
+            if not identifier:
+                message = OPDSMessage(
+                    urn, INVALID_URN.status_code, INVALID_URN.detail
+                )
+            else:
+                description = "Already in catalog"
+                if identifier not in collection.catalog:
+                    collection.catalog_identifier(self._db, identifier)
+                    description = "Successfully added"
```

Comment:
I don't think it matters very much, but the response code in this case should be 201 ("Created"), not 200 ("OK").